pipeline/extract/extract_gog.py

```diff
@@ -1,56 +1,122 @@
-    pub = game_soup.find_all('a', class_='details__link')[-2].text
-    date = game_soup.find_all(
-        'div', class_='details__content table__row-content')
-    date2 = game_soup.find_all(
-        'span', {'ng-cloak': ''})
-
-    date3 = game_soup.find_all(
-        'span', class_='ng-binding')
-    # print(date2)
-    # print(date3)
-    '''<div class="details__content table__row-content"><span ng-cloak="">{{'2024-04-19T00:00:00+03:00' | date: 'longDate' : ' +0300 ' }}</span></div>,'''
-
-    rating = game_soup.find_all(
-        'span', class_='average-item__value ng-binding')
-    print(rating)
-    # print(dev, pub)
+    return [t.text for t in tags]
+
+
+def get_price(game_dict: dict) -> float:
+    '''Returns the current price of the game as a float.'''
+    return float(game_dict['offers']['price'])
+
+
+def get_rating(game_dict: dict) -> float:
+    '''Returns the average rating for a game as a float,
+    out of 5.'''
+    if 'aggregateRating' not in game_dict.keys():
+        return None
+    return float(game_dict['aggregateRating']['ratingValue'])
+
+
+def get_release_date(game_dict: dict) -> datetime:
+    '''Returns a datetime object of the release day.'''
+    return datetime.strptime(game_dict['releaseDate'][:-6], '%Y-%m-%dT%H:%M:%S')
+
+
+def get_platform_ids(platform_str: str) -> list:
+    '''Returns a list of platform IDs given a
+    string of playable platforms.'''
+    id_list = []
+    if 'windows' in platform_str:
+        id_list.append(1)
+    if 'osx' in platform_str:
+        id_list.append(2)
+    if 'linux' in platform_str:
+        id_list.append(3)
+    return id_list
+
+
+def get_title(game_soup_small) -> str:
+    '''Returns a string of the game's name.'''
+    return game_soup_small.find(
+        'div', class_='product-tile__title')['title']
 
 
 if __name__ == "__main__":
 
     load_dotenv()
 
-    res = req.get(ENV["SCRAPING_URL"])
-    soup = BeautifulSoup(res.text, features="html.parser")
+    response_all_games = req.get(ENV["SCRAPING_URL"], timeout=5)
+    soup = BeautifulSoup(response_all_games.text, features="html.parser")
     soup = soup.findAll('product-tile', class_='ng-star-inserted')
-    for game in soup[:1]:
-        title_object = game.find(
-            'div', class_='product-tile__title')['title']
-        price = game.find('span', class_='final-value ng-star-inserted')
-        price = price.text if price is not None else '$0'
+    for game in soup[:10]:
         address = game.find(
             'a', class_='product-tile product-tile--grid')['href']
-        website_id = 2
-        print(title_object, price, address, website_id)
-        scrape_from_game_page(address)
+        response = req.get(address, timeout=5)
+        game_data = BeautifulSoup(response.text, features="html.parser")
+        game_json = get_json(game_data)
+        link = get_detail_links(game_data)
+        print([get_title(game), get_price(game_json), get_developer(link),
```

Comment:
why are you printing :(

Reply:
Good point I will break this into a function and return!